### DISCUSSION TITLE
Enable navigation blocker based on "next" route

### DISCUSSION THREAD

**Discussion by chrisbenincasa** [TanStack/router#1733] [Q&A]
I'm wondering if there is a way to recreate the functionality from react-router v6's useBlocker, which can determine blocked state based on the "next" location (about to transition): https://reactrouter.com/en/main/hooks/use-blocker.

I've gotten close with watching changes in location.pathname - but this change seems to fire after the transition has begun and after the blocker has permitted the action. Is there a way to preempt the transition and update the blocking condition based on the upcoming path?

**Comment by salemlf** [comment]
There's a PR in progress to implement this: https://github.com/TanStack/router/pull/1790

I'm also needing the same behavior and haven't been able to find any hacky workarounds in the meantime
